### PR TITLE
Fix selftest

### PR DESCRIPTION
**Pull request by alexfmpe**

### CODE REVIEW

File: lib/selftest/src/Obelisk/SelfTest.hs [hunk 1]
```diff
@@ -166,9 +165,6 @@ main = do
           it ("can enter "    <> shell) $ inTmpObInit $ \_ -> inShell "exit"
           it ("can build in " <> shell) $ inTmpObInit $ \_ -> inShell $ "cabal new-build --" <> fromString compiler <> " all"
 
-        it "can build reflex project" $ inTmpObInit $ \_ -> do
```
Comment (alexfmpe) [inline lib/selftest/src/Obelisk/SelfTest.hs]: Fails with either `error: Android builds are not supported on this platform.` or `error: iOS builds are not supported on this platform.`.

File: default.nix [hunk 2]
```diff
@@ -128,7 +128,7 @@ in rec {
   obelisk = ghcObelisk;
   obeliskEnvs = ghcObeliskEnvs;
   command = ghcObelisk.obelisk-command;
-  shell = pinBuildInputs "obelisk-shell" ([command] ++ commandRuntimeDeps pkgs) [];
```
Comment (alexfmpe) [inline default.nix]: Not sure what this `[]` was supposed to do, but it causes `error: attempt to call something which is not a function but a set`

Comment (3noch) [reply default.nix]: Probably some older nixpkgs thing that went away when bumped.

File: lib/selftest/src/Obelisk/SelfTest.hs (hunk 1 above)
Comment (3noch) [reply lib/selftest/src/Obelisk/SelfTest.hs]: Yeah this is not my favorite design that `nix-build` doesn't work without `-A`. Perhaps we can add a `-A all-supported-platforms` or something that just filters out anything that doesn't get through `tryEval`.

Comment (3noch) [reply lib/selftest/src/Obelisk/SelfTest.hs]: Or something much less hacky...

Comment (3noch) [reply lib/selftest/src/Obelisk/SelfTest.hs]: I think that would be a better solution that removing this test case. What do you think?

**Comment by ali-abrar** [comment]
Thanks

**Pull request merged**